src/main/java/org/assertj/core/api/AbstractLongAdderAssert.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.assertj.core.data.Offset;
+import org.assertj.core.data.Percentage;
+import org.assertj.core.internal.ComparatorBasedComparisonStrategy;
+import org.assertj.core.internal.Longs;
+import org.assertj.core.util.CheckReturnValue;
+import org.assertj.core.util.VisibleForTesting;
+
+import java.util.Comparator;
+import java.util.concurrent.atomic.LongAdder;
+
+import static org.assertj.core.error.ShouldHaveValue.shouldHaveValue;
+import static org.assertj.core.error.ShouldNotContainValue.shouldNotContainValue;
+
+/**
+ * Base class for all implementations of assertions for {@link LongAdder}s.
+ *
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a href="http://bit.ly/1IZIRcY"
+ *               target="_blank">Emulating 'self types' using Java Generics to simplify fluent API implementation</a>&quot;
+ *               for more details.
+ *
+ * @author Grzegorz Piwowarek
+ * @since 3.16.0
+ */
+public class AbstractLongAdderAssert<SELF extends AbstractLongAdderAssert<SELF>> extends AbstractAssert<SELF, LongAdder> implements NumberAssert<SELF, Long>, ComparableAssert<SELF, Long> {
+
+  @VisibleForTesting
+  Longs longs = Longs.instance();
+
+  public AbstractLongAdderAssert(LongAdder longAdder, Class<?> selfType) {
+    super(longAdder, selfType);
+  }
+
+  /**
+   * Verifies that the actual sum has the given value.
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * // assertion will pass
+   * LongAdder actual = new LongAdder();
+   * actual.add(42);
+   * assertThat(actual).hasValue(42);
+   *
+   * // assertion will fail
+   * assertThat(actual).hasValue(0);</code></pre>
+   *
+   * @param expectedValue the value not expected .
+   *
+   * @return {@code this} assertion object.
+   *
+   * @throws AssertionError if the actual adder is {@code null}.
+   */
+  public SELF hasValue(long expectedValue) {
```

Comment:
Let's try to shoot on the remaining `hasValue`/`doesNotHaveValue` pair 🙂
Would it be too aggressive to replace them with `isEqualTo(Long)`/`isNotEqualTo(Long)` similarly to `AbstractLongAssert`, which delegate to `longs.assertEqual()`/`longs.assertNotEqual()` respectively?

@joel-costigliola thoughts?

Reply:
As discussed with @joel-costigliola, let's keep `hasValue`/`doesNotHaveValue`.